File: frontend/src/types/training.ts
export type TrainingSummary = {
  total_runs: number
  ac_count: number
  wrong_count: number
  submit_count: number
  strongest_tag: string | null
  main_error_type: 'AC' | 'WA' | 'RE' | 'CE' | 'TLE' | 'PENDING' | null
}

export type TrainingRecentItem = {
  submission_id: number
  problem_id: number
  title: string
  company: string
  language: 'Python' | 'C++' | 'Java'
  run_type: 'run' | 'submit'
  verdict: 'AC' | 'WA' | 'RE' | 'CE' | 'TLE' | 'PENDING'
  created_at: string
}

export type TrainingErrorBucket = {
  verdict: 'AC' | 'WA' | 'RE' | 'CE' | 'TLE' | 'PENDING'
  count: number
}

export type TrainingRecommendation = {
  problem_id: number
  title: string
  company: string
  tags: string[]
  recommendation_reason: string
}

export type TrainingOverviewResponse = {
  summary: TrainingSummary
  recent_items: TrainingRecentItem[]
  error_buckets: TrainingErrorBucket[]
  recommendations: TrainingRecommendation[]
}

File: backend/src/schemas/training.py
from __future__ import annotations

from pydantic import BaseModel, Field

from schemas.submissions import Language, RunType, Verdict


class TrainingSummary(BaseModel):
    total_runs: int = Field(ge=0)
    ac_count: int = Field(ge=0)
    wrong_count: int = Field(ge=0)
    submit_count: int = Field(ge=0)
    strongest_tag: str | None = None
    main_error_type: Verdict | None = None


class TrainingRecentItem(BaseModel):
    submission_id: int = Field(ge=1)
    problem_id: int = Field(ge=1)
    title: str
    company: str
    language: Language
    run_type: RunType
    verdict: Verdict
    created_at: str


class TrainingErrorBucket(BaseModel):
    verdict: Verdict
    count: int = Field(ge=0)


class TrainingRecommendation(BaseModel):
    problem_id: int = Field(ge=1)
    title: str
    company: str
    tags: list[str]
    recommendation_reason: str


class TrainingOverviewResponse(BaseModel):
    summary: TrainingSummary
    recent_items: list[TrainingRecentItem]
    error_buckets: list[TrainingErrorBucket]
    recommendations: list[TrainingRecommendation]
